scripts/test_e2e.sh

```
@@ -0,0 +1,144 @@
+case "$TARGET_DIR" in /*) ;; *) TARGET_DIR="$(pwd)/$TARGET_DIR" ;; esac
+# Escape every ERE metacharacter so `pkill -f` matches the path literally.
+TARGET_RE="$(printf '%s' "$TARGET_DIR" | sed 's/[][(){}.^$*+?|\\]/\\&/g')"
+LOG=""
+cleanup() {
+  pkill -TERM -f "$TARGET_RE/.*/deps/(e2e|integration)-" 2>/dev/null
+  if pgrep -f "$TARGET_RE/.*/deps/(e2e|integration)-" >/dev/null 2>&1; then
+    sleep 1
+    pkill -KILL -f "$TARGET_RE/.*/deps/(e2e|integration)-" 2>/dev/null
+  fi
+  [ -n "$LOG" ] && rm -f "$LOG"
+}
```

Shellcheck (0.11.0)

[info] This function is never invoked. Check usage (or ignored if invoked indirectly).

(SC2329)